fdm-core/src/authorization.d.ts

```diff
@@ -0,0 +1,33 @@
+import type * as authZSchema from "./db/schema-authz"
+
+export type Resource =
+    | "user"
+    | "organization"
+    | "farm"
+    | "field"
+    | "cultivation"
+    | "soil_analysis"
+    | "fertilizer_application"
+    | "soil"
+    | "harvesting"
+export type Role = "owner" | "advisor" | "researcher"
```

Comment:
catalogues are independently checked for authorisation?

Reply:
Currently catalogues are set at application level and not per user. However, this needs to be implemented before alpha release in a next PR

Reply:
okido. seems fine to me.